### ISSUE TITLE
[Bug] Comments in global .yarnrc.yml break yarn install

### ISSUE THREAD

**Issue by dradetsky** [yarnpkg/berry#2270]
- [ ] I'd be willing to implement a fix

**Describe the bug**

I was trying to get berry working for me & configuring it to my taste. I had put some lines in my `~/.yarnrc.yml`, including:

```
initVersion: '0.0.0'
```

Which worked on the version of yarn installed by following the instructions at https://yarnpkg.com/getting-started/install

I then found something I wanted to change, cloned the repo, and started by trying to build & run tests before I made a change. However, the first thing that happend was

```
% yarn install
Usage Error: Unrecognized or legacy configuration settings found: initVersion - run "yarn config -v" to see the list of settings supported in Yarn (in /home/dmr/.yarnrc.yml)

$ yarn install [--json] [--immutable] [--immutable-cache] [--check-cache] [--inline-builds] [--skip-builds]
```

So this was initially quite confusing. I'm guessing what happened was that you guys deprecated that option in favor of allowing `initFields.version` (this was what I tried first before seeing `initVersion` in the output of `yarn config`) as in #2230.

But at this point the natural response is to comment that line. However, with my `~/.yarnrc.yml` as

```
# initVersion: '0.0.0'
```

the result is

```
% yarn install
TypeError: Cannot read property 'yarnPath' of null
    at loadRcFile (/home/dmr/.npm/lib/node_modules/yarn/lib/cli.js:56945:49)
    at /home/dmr/.npm/lib/node_modules/yarn/lib/cli.js:56916:14
    at /home/dmr/.npm/lib/node_modules/yarn/lib/cli.js:101331:14
    at Array.map (<anonymous>)
    at parseRcPaths (/home/dmr/.npm/lib/node_modules/yarn/lib/cli.js:101329:78)
    at Object.findRc (/home/dmr/.npm/lib/node_modules/yarn/lib/cli.js:101343:10)
    at getRcConfigForCwd (/home/dmr/.npm/lib/node_modules/yarn/lib/cli.js:56915:74)
    at /home/dmr/.npm/lib/node_modules/yarn/lib/cli.js:92694:56
    at Generator.next (<anonymous>)
    at step (/home/dmr/.npm/lib/node_modules/yarn/lib/cli.js:310:30)
```

Which definitely doesn't seem right. I haven't tested this extensively, but my guess is that berry doesn't handle an empty config file properly. If I put anything else uncommented in it, such as

```
# initVersion: '0.0.0'
preferInteractive: false
```

then `yarn install` works.

**To Reproduce**

imo the above should be sufficient to reproduce. However, I'll take a crack at using sherlock (this probably wont work)

<details>
  <summary>Reproduction</summary>

  ```js repro
  // Sherlock reproduction. For instance:
const { promises: { writeFile } } = require(`fs`);

await writeFile(
  '~/.yarnrc.yml',
  "initVersion: '0.0.0'"
)
await expect(yarn('version')).resolves.toBeTruthy();

await writeFile(
  '~/.yarnrc.yml',
  "# initVersion: '0.0.0'"
)
await expect(yarn('version')).resolves.toBeTruthy();

await writeFile(
  '~/.yarnrc.yml',
  `# initVersion: '0.0.0'
   preferInteractive: false`
)
await expect(yarn('version')).resolves.toBeTruthy();
```
</details>

**Environment if relevant (please complete the following information):**

```
% node -v
v15.5.0
% uname -a
Linux lapl 5.9.14-arch1-1 #1 SMP PREEMPT Sat, 12 Dec 2020 14:37:12 +0000 x86_64 GNU/Linux
```

yarn version: 2460f384df75fef59b1ba4efee002c3af8a369ef

**Comment by yarnbot** [comment]
The reproduction case in your issue seems broken (ie it neither pass nor fail due to throwing an unmanaged exception):
```
Error: ENOENT: no such file or directory, open '~/.yarnrc.yml'
```

Remember: any non-Jest exceptions will cause the test to be reported as broken. If you expect something to pass without throwing, you must wrap it into something like `await expect(...).resolves.toBeTruthy()`. If you instead expect something to throw, you need to wrap it into `await expect(...).rejects.toThrow()`.

**Comment by merceyz** [comment]
That seems like v1 throwing an error

**Comment by arcanis** [comment]
> Usage Error: Unrecognized or legacy configuration settings found: initVersion - run "yarn config -v" to see the list of settings supported in Yarn (in /home/dmr/.yarnrc.yml)

This setting got removed in https://github.com/yarnpkg/berry/pull/2264 (since we've started work on 3.x, using builds from master may contain a few breaking changes like this from time to time).

**Comment by merceyz** [comment]
Though I can confirm Yarn v1 throws if the `.yarnrc.yml` only contains comments

**Comment by arcanis** [comment]
As for:

```
TypeError: Cannot read property 'yarnPath' of null
    at loadRcFile (/home/dmr/.npm/lib/node_modules/yarn/lib/cli.js:56945:49)`
```

It seems v1 indeed - I think it's because it doesn't handle the case where the yarnrc file is completely empty. Just remove the file altogether and it should work (although in your case you probably just want to use `initFields` instead).

**Comment by dradetsky** [comment]
> Just remove the file altogether and it should work

@arcanis It's not that I'm unable to work around the issue. I was able to work around it quite happily when I submitted the report. The point was that the error was one which I thought was likely to be confusing for new users of yarn/berry, so maybe something you ought to fix. I might have seen the two errors one after the other (as in fact I did) and thought that berry was still unstable trash & that I should not follow your advice & upgrade to berry now.

I don't yet know enough about the project to know whether this is due to a v1 issue (it seems weird to me that the v1 code would read the v2 config file, but whatever), but if it's relevant

```
% yarn --version
1.22.4
````
